pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -1274,8 +1279,9 @@ pub mod pallet {
 
 			Pools::<T>::remove(&pool_id);
 
+			let hold_reason = T::HoldReason::from((pool_id.clone(), pool_details.owner.clone()));
 			T::DepositCurrency::release(
-				&T::RuntimeHoldReason::from(HoldReason::Deposit),
+				&T::RuntimeHoldReason::from(hold_reason),
```

Comment:
given that the hold reason is unique to a pool, we wouldn't even have to store the amount held in this pallet any more, right? We could just release any amount held under this reason?

Reply:
Exactly. That's the whole point why I tried to make the deposit storage pallet work. Everything will be stored in there under the `BondedCurves` namespace and `(pool_id, owner)` key, or even simply `pool_id` now that I think about it, if there is a single deposit taken for each pool and each pool ID is unique.

Reply:
Plus you don't need to store the owner anymore, as you can use the `InspectHold` trait for a specific reason and user, to know whether that user has a reserved deposit for the reason or not.